base_layer/wallet_ffi/src/lib.rs

```diff
@@ -4189,6 +4213,24 @@ pub unsafe extern "C" fn wallet_get_utxos(
     }
 }
 
+/// Frees memory for a `TariOutputs`
+///
+/// ## Arguments
+/// `x` - The pointer to `TariOutputs`
+///
+/// ## Returns
+/// `()` - Does not return a value, equivalent to void in C
+///
+/// # Safety
+/// None
+#[no_mangle]
+pub unsafe extern "C" fn destroy_tari_outputs(x: *mut TariOutputs) {
+    if !x.is_null() {
+        Vec::from_raw_parts((*x).ptr, (*x).len, (*x).cap);
+        Box::from_raw(x);
```

Comment:
Interesting that you construct and drop the Vec AND Box - you would think that dropping the Vec is enough? Since the data is dropped in the line above (I think that is how rust does it if you don't bind to a variable) would Box::from_raw now not point to unallocated data?

Reply:
I'm not sure I completely understood your point.

Reply:
Boxed value is a struct containing internal pointer, its length and capacity. I'm following the [docs on into_raw_parts](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.into_raw_parts). I'm not sure that reconstructing and dropping the box would release the memory allocated for the internal array pointer, would it?

Reply:
Only saw your comments now - questions for my own interest, trying to understand as I'm not an `unsafe` expert 😄  

Thanks, makes sense to me now